content/blog/gpg-signed-releases/index.md

```diff
@@ -0,0 +1,121 @@
+---
+title: Verifying GPG signatures for Temurin downloads
+date: "2022-07-27T16:00:00+00:00"
+author: sxa
+description: How to verify GPG signatures of the Temurin artifacts using the Eclipse public key
+tags:
+  - Temurin
+  - Security
+---
+
+With the latest releases from July 2022 (8u342, 11.0.16, 17.0.4 and 18.0.2)
+we provide GPG signatures along with our releases which you can use to
+verify that the downloads you have are genuine and have not been tampered
+with since we built them.  This gives a quick crib sheet of what you need to
+do to verify the downloads.
+
+## What is GPG signing?
+
+## You are already providing SHA256 checksums. Why do I need this?
+
+While going into the details of public key encryption is beyond the scope of
+this article, if there was a man in the middle (MITM) attach that resulted
+in the downloads you receive being compromised, then the corresponding SHA
+checksums could also be compromised.  GPG signing avoids this issue by using
+a separately certified signature which you can initially trust and then verify
+subsequent downloads.
+
+## What do I need to do to obtain the signatures?
+
+The link to the signatures is provided in the metadata for our releases.  In
+the following examples I will use JDK17 but you can change the calls
+appropriately for other versions:
+
+The data file that contains the URL of the signatures can be obtained by
+downloading the json as follows:
+
+`curl 'https://api.adoptium.net/v3/assets/feature_releases/17/ea?page=0&page_size=1' > adopt.json`
+
+Once you've downloaded that you need to extract the `link` and
+`signature_link` entries using your preferred JSON parsing tool.  The
+information you want is in the `binaries[0].package.link` and
+`binaries[0].package.signature_link` section of the file for the most recent
+version.
+
+For automation purposes, I will give an example using the
+[jq](https://stedolan.github.io/jq/) command line JSON parsing tool to pull
+out the information for Linux/x64 and download the required artifacts - the
+product download and the GPG signature link - using the `curl` command
+(Examples for linux/x64, change accordingly for other platforms)
+
+```
+jq '.[0].binaries[] | select(.architecture=="x64") | select (.os=="linux") | select (.image_type=="jdk") .package.link' adopt.json | xargs curl -LO
+jq '.[0].binaries[] | select(.architecture=="x64") | select (.os=="linux") | select (.image_type=="jdk") .package.signature_link' adopt.json | xargs curl -LO
+```
+
+Note that as mentioned in the previous section, we have previously also
+provided the sha256sums which can be obtained from the JSON file with
+
+```
+jq '.[0].binaries[] | select(.architecture=="x64") | select (.os=="linux") | select (.image_type=="jdk") .package.checksum' adopt.json
+```
+
+If you have not used them before, The SHA checksums can be verified against
+the output from running the `sha256sum` command against the binary artifact.
```

Comment:
On Windows use `certUtil -hashfile file SHA256`, on Linux use `sha256sum file`, and on macOS use `shasum -a 256 file`.

Reply:
All the windows systems I have had sha256sum on them, but since that's not a default and won't be available on everyone's systems, I've added those into three bullet points.